src/v/cluster/topics_frontend.cc

```diff
@@ -1284,4 +1284,48 @@ ss::future<std::error_code> topics_frontend::decrease_replication_factor(
       _stm, _features, _as, std::move(cmd), timeout);
 }
 
+allocation_request make_allocation_request(
+  model::partition_id pid,
+  topic_configuration topic_cfg,
+  const std::vector<model::node_id>& new_replicas) {
+    allocation_request req(get_allocation_domain(topic_cfg.tp_ns));
+    req.partitions.reserve(1);
+    allocation_constraints constraints;
+    constraints.hard_constraints.push_back(
+      ss::make_lw_shared<hard_constraint_evaluator>(on_nodes(new_replicas)));
+    req.partitions.emplace_back(
+      pid, topic_cfg.replication_factor, std::move(constraints));
+    return req;
+}
+
+ss::future<result<std::vector<partition_assignment>>>
+topics_frontend::generate_reassignments(
```

Comment:
would be great to get a review on these changes to topics frontend @ztlpn

Reply:
I am wondering if it would be cleaner to expose a `move_partition_replicas` override that would accept `std::vector<model::node_id>` instead of `std::vector<model::broker_shard>` this way we wouldn't have to leak allocation details outside to Kafka layer ?

Reply:
We could then make it more encapsulated

Reply:
good idea. I'll change